### ISSUE TITLE
Q/B: unsustainable results

### ISSUE THREAD

**Issue by mr-moon** [karma-runner/karma#1172]
I'm having very unpredictable results here.

My script depends on other scripts. Those dependencies are provided asynchronously. But in Karma, I cannot get sustained results. onload event doesn't get always fired. And by that I mean that without any changes to the code, I `karma start`-ed the testing and it all worked — I saw that `loaded` output! But the other test I ran — it did not.

This is the simple code I have in my spec (FeedServiceTest.js):
```javascript
var el = document.createElement('script');
el.onload = el.onerror = el.onreadystatechange = function () {
    console.log('loaded');
    // descibe('component ...
};
el.async = 1;
el.src = '//cdnjs.cloudflare.com/ajax/libs/script.js/2.5.7/script.min.js?r='+Math.random();
var head = document.getElementsByTagName('head')[0];
head.insertBefore(el, head.lastChild);
```
And the funny thing, I cannot get 'loaded' logged. I see that file has been loaded by the borwser. It's clearly visible in the Chrome DevTools. But onload isn't getting fired.

I also tried to listen to load event via $(el).on('load', ... ); — same problem. Sometimes I see that request has been cancelled.

Just noticed, that by pressing "debug" button in Chrome — I started to see those 'loaded' logs.

Am I doing anything wrong?

```
Chrome 37.0.2062 (Mac OS X 10.9.4)
Karma version: 0.12.16
node v0.10.31
```

Here is the _expected_ log:
```
INFO [karma]: Karma v0.12.16 server started at http://localhost:9876/
INFO [launcher]: Starting browser Chrome
INFO [Chrome 37.0.2062 (Mac OS X 10.9.4)]: Connected on socket Ag7u2dFlHPFR9tEdO2IM with id 16775281
Chrome 37.0.2062 (Mac OS X 10.9.4): Executed 0 of 0 ERROR (0.002 secs / 0 secs)
INFO [watcher]: Changed file "/tests/unit/FeedServiceTest.js".
LOG: 'loaded'
Chrome 37.0.2062 (Mac OS X 10.9.4): Executed 0 of 0 ERROR (0.4 secs / 0 secs)
INFO [watcher]: Changed file "/tests/unit/FeedServiceTest.js".
LOG: 'loaded'
```
And here's the one I get the most:
```
INFO [karma]: Karma v0.12.16 server started at http://localhost:9876/
INFO [launcher]: Starting browser Chrome
INFO [Chrome 37.0.2062 (Mac OS X 10.9.4)]: Connected on socket F_y2RtKC4-zlcJngNqTF with id 13037247
Chrome 37.0.2062 (Mac OS X 10.9.4): Executed 0 of 0 ERROR (0.312 secs / 0 secs)
INFO [watcher]: Changed file "/tests/unit/FeedServiceTest.js".
Chrome 37.0.2062 (Mac OS X 10.9.4): Executed 0 of 0 ERROR (0.242 secs / 0 secs)
INFO [watcher]: Changed file "/tests/unit/FeedServiceTest.js".
Chrome 37.0.2062 (Mac OS X 10.9.4): Executed 0 of 0 ERROR (0.312 secs / 0 secs)
INFO [watcher]: Changed file "/tests/unit/FeedServiceTest.js".
```

By `Changed` Karma means adding and removing some whitespaces .

**Comment by ThiagoMiranda** [comment]
@mr-moon I'm having the same issue here ( http://stackoverflow.com/questions/29371180/how-to-test-a-file-that-dynamically-loads-other-javascript-files ).
Did you solved this?

**Comment by mr-moon** [comment]
@ThiagoMiranda nope. I'm waiting for karma to support `node 0.12`.

**Comment by ThiagoMiranda** [comment]
@mr-moon Well I'm running it on v0.12.0.
It seems that is a problem with the PhantomJS itself that maybe it's fixed on the 2.0 version. The problem is that the karma-phantomjs-launcher only supports the 1.x version.
Running the Firefox test solved the problem ( this one, at least =p )

**Comment by Dignifiedquire** [comment]
Not related to karma itself..There is also a way to use phantomjs 2.0 with the launcher now. So closing this issue
